barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/goblin_translator_circuit_builder.hpp

```diff
@@ -8,13 +8,71 @@
  * @copyright Copyright (c) 2023
  *
  */
+#include "barretenberg/common/constexpr_utils.hpp"
 #include "barretenberg/ecc/curves/bn254/fq.hpp"
+#include "barretenberg/numeric/uint256/uint256.hpp"
 #include "barretenberg/proof_system/arithmetization/arithmetization.hpp"
+#include "barretenberg/proof_system/op_queue/ecc_op_queue.hpp"
 #include "circuit_builder_base.hpp"
 #include <array>
 #include <cstddef>
+#include <cstdlib>
+#include <iterator>
+#include <tuple>
 namespace proof_system {
-class GoblinTranslatorCircuitBuilder : CircuitBuilderBase<arithmetization::GoblinTranslator> {
+/**
+ * @brief GoblinTranslatorCircuitBuilder creates a circuit that evaluates the correctness of the evaluation of
+ * EccOpQueue in Fq while operating in the Fr scalar field
+ *
+ * @details Goblin Translator Circuit Builder builds a circuit the purpose of which is to calculate the batched
+ * evaluation of 5 polynomials in non-native field represented through coefficients in 4 native polynomials (op,
+ * x_lo_y_hi, x_hi_z_1, y_lo_z_2):
+ *
+ * OP | X_LO | X_HI | Y_LO
+ * 0  | Y_HI | Z1   | Z2
+ *
+ *  OP is supposed to be { 0, 1, 2, 3, 4, 8 }. X_LO and Y_LO need to be < 2¹³⁶, X_HI and Y_LO < 2¹¹⁸, Z1 and Z2 < 2¹²⁸.
+ *  X_* and Y_* are supposed to be the decompositions of bn254 base fields elements P.x and P.y and are split into two
+ * chunks each because the scalar field we are operating on can't fit them
+ *
+ * Goblin Translator calculates the result of evaluation of a polynomial op + P.x⋅v +P.y⋅v² + z1 ⋅ v³ + z2⋅v⁴ at the
```

Comment:
This comment block is very helpful. My only request is an explicit statement about the meaning of r and p. They are the scalar and base field of bn254? Or equivalently the moduli of Fr and Fq? I know that the translator currently only makes sense over bn254 but since Grumpkin is also floating around a lot I think its nice to be explicit

Reply:
Added. r is modulus of Fr (scalar), p is modulus of Fq (base)